src/modules/Visualizer/LayersTree/LayersTreeItem/LayersTreeExclusiveItemsList/Selector/Selector.js

```diff
@@ -8,81 +8,132 @@ import translateMock from '../../../../../../utils/translate';
 
 import './styles.scss';
 
-export class Selector extends React.Component {
-  static propTypes = {
-    selectors: PropTypes.arrayOf(PropTypes.shape({
-      label: PropTypes.string,
-      name: PropTypes.string.isRequired,
-      values: PropTypes.arrayOf(PropTypes.shape({
-        label: PropTypes.string.isRequired,
-        value: PropTypes.string.isRequired,
-      })).isRequired,
-    })).isRequired,
-    layers: PropTypes.arrayOf(PropTypes.shape({
-      selectorKey: PropTypes.object,
-    })).isRequired,
-    activeLayer: PropTypes.shape({
-      selectorKey: PropTypes.object,
-    }).isRequired,
-    onChange: PropTypes.func,
-    translate: PropTypes.func,
-  }
-
-  static defaultProps = {
-    onChange () {},
-    translate: translateMock({
-      'terralego.visualizer.layerstree.group.selector': 'No layer found',
-    }),
-  }
-
-  state = {}
-
-  onChange = name => value => {
-    const { selectors, layers, activeLayer: { selectorKey }, onChange } = this.props;
-    const newKey = { ...selectorKey, [name]: value };
-    const newActiveLayerIndex = layers.findIndex(({ selectorKey: layerSelectorKey }) =>
+const SelectorSelect = ({
+  label,
+  name,
+  values,
+  currentValue,
+  onChange,
+  ...props
+}) => {
+  const handleChange = React.useCallback(
+    value => onChange(name, value),
+    [name, onChange],
+  );
+
+  /**
+   * Get selected value from all values.
+   */
+  const getSelectedValue = React.useMemo(() => {
+    if (values.length) {
+      const found = values.find(({ value }) => currentValue === value);
+      if (!found) {
+        return values[0].value;
+      }
+      return found.value;
```

Comment:
Not sure if this is better, but it could be simplify like this
```js
const [{ value } = values[0]] = values.filter(({ value }) => currentValue === value);
return value
```

Reply:
`.filter()` will iterate over the entire array, when `.find()` stop as soon as callback return a truthy value...

Reply:
True 😉

Reply:
This would achieve the same and also keep the `.find()`

```suggestion
      const { value } = values.find(({ value }) => currentValue === value) || values[0];
      return value;
```

Reply:
Done